go-controller/pkg/cni/cni_dpu.go

```diff
@@ -22,13 +22,16 @@ func (pr *PodRequest) updatePodDPUConnDetailsWithRetry(kube kube.Interface, podL
 		cpod := pod.DeepCopy()
 		cpod.Annotations, err = util.MarshalPodDPUConnDetails(cpod.Annotations, dpuConnDetails, pr.nadName)
 		if err != nil {
+			if util.IsAnnotationAlreadySetError(err) {
```

Comment:
@cathy-zhou will the annotation ever change? eg, do we need to just ignore the error if the annotation is already set and overwrite the existing one?

Reply:
we could, it is just to save a k8s-api-server trip, also descrease the chance of `Conflict` error. As I understand, k8s-api-server interaction can sometimes be heavy.